VirtoCommerce.CatalogPersonalizationModule.Data/Services/PersonalizationService.cs

```diff
@@ -7,117 +8,198 @@
 
                 var query = repository.TaggedItems;
 
-				if (!criteria.EntityIds.IsNullOrEmpty())
-				{
-					query = query.Where(x => criteria.EntityIds.Contains(x.ObjectId));
-				}
-
-				if (!string.IsNullOrWhiteSpace(criteria.EntityType))
-				{
-					query = query.Where(x => x.ObjectType == criteria.EntityType);
-				}
-
-				if (!criteria.Ids.IsNullOrEmpty())
-				{
-					query = query.Where(x => criteria.Ids.Contains(x.Id));
-				}
-
-				if (criteria.ChangedFrom.HasValue)
-				{
-					query = query.Where(x => x.ModifiedDate.HasValue && x.ModifiedDate.Value.Date >= criteria.ChangedFrom.Value.Date);
-				}
-
-				var sortInfos = criteria.SortInfos;
-				if (sortInfos.IsNullOrEmpty())
-				{
-					sortInfos = new[] { new SortInfo { SortColumn = ReflectionUtility.GetPropertyName<TaggedItem>(x => x.Label) } };
-				}
-				query = query.OrderBySortInfos(sortInfos);
-
-				retVal.TotalCount = query.Count();
-				query = query.Skip(criteria.Skip).Take(criteria.Take);
-
-				var ids = query.Select(x => x.Id).ToArray();
-				retVal.Results = GetTaggedItemsByIds(ids, criteria.ResponseGroup).AsQueryable().OrderBySortInfos(sortInfos).ToList();
-			}
-			return retVal;
-		}
-
-		public void DeleteTaggedItems(string[] ids)
-		{
-			using (var repository = _repositoryFactory())
-			{
-				repository.DeleteTaggedItems(ids);
-				CommitChanges(repository);
-			}
-		}
-        	
-	}
+                if (!criteria.EntityIds.IsNullOrEmpty())
+                {
+                    query = query.Where(x => criteria.EntityIds.Contains(x.ObjectId));
+                }
+
+                if (!string.IsNullOrWhiteSpace(criteria.EntityType))
+                {
+                    query = query.Where(x => x.ObjectType == criteria.EntityType);
+                }
+
+                if (!criteria.Ids.IsNullOrEmpty())
+                {
+                    query = query.Where(x => criteria.Ids.Contains(x.Id));
+                }
+
+                if (criteria.ChangedFrom.HasValue)
+                {
+                    query = query.Where(x => x.ModifiedDate.HasValue && x.ModifiedDate.Value.Date >= criteria.ChangedFrom.Value.Date);
+                }
+
+                var sortInfos = criteria.SortInfos;
+                if (sortInfos.IsNullOrEmpty())
+                {
+                    sortInfos = new[] { new SortInfo { SortColumn = ReflectionUtility.GetPropertyName<TaggedItem>(x => x.Label) } };
+                }
+                query = query.OrderBySortInfos(sortInfos);
+
+                retVal.TotalCount = query.Count();
+                query = query.Skip(criteria.Skip).Take(criteria.Take);
+
+                var ids = query.Select(x => x.Id).ToArray();
+                retVal.Results = GetTaggedItemsByIds(ids, criteria.ResponseGroup).AsQueryable().OrderBySortInfos(sortInfos).ToList();
+
+                if (criteria.ResponseGroup.Contains(TaggedItemResponseGroup.WithInheritedTags.ToString()) &&
```

Comment:
It is the really weird way to check flags, why you do this by this way?  We have a lot of example how to do this? could you check? https://github.com/VirtoCommerce/vc-module-catalog/blob/dev/VirtoCommerce.CatalogModule.Data/Repositories/CatalogRepositoryImpl.cs#L275